pkg/operator/ceph/spec/spec.go

```diff
@@ -379,6 +380,49 @@ func ChownCephDataDirsInitContainer(
 	}
 }
 
+// GenerateMinimalCephConfInitContainer returns an init container that will generate the most basic
+// Ceph config for connecting non-Ceph daemons to a Ceph cluster (e.g., nfs-ganesha). Effectively
+// what this means is that it generates '/etc/ceph/ceph.conf' with 'mon_host' populated and a
+// keyring path associated with the user given. 'mon_host' is determined by the 'ROOK_CEPH_MON_HOST'
+// env var present in other Ceph daemon pods, and the keyring is expected to be mounted into the
+// container with a Kubernetes pod volume+mount.
+func GenerateMinimalCephConfInitContainer(
+	username, keyringPath string,
+	containerImage string,
+	volumeMounts []v1.VolumeMount,
+	resources v1.ResourceRequirements,
+	securityContext *v1.SecurityContext,
+) v1.Container {
+	cfgPath := cephconfig.DefaultConfigFilePath()
+	// Note that parameters like $(PARAM) will be replaced by Kubernetes with env var content before
+	// container creation.
+	confScript := `
+set -xEeuo pipefail
```

Comment:
I believe some options (such as `pipefail`) are not really needed, although they don't hurt...

Reply:
I'd rather leave it in case someone comes through later and adds piped commands.